### DISCUSSION TITLE
JDBC MessageStore with several number of sessions / connection pooling library

### DISCUSSION THREAD

**Discussion by chrjohn** [quickfix-j/quickfixj#373] [Q&A] [answered]
Hi,

does anyone have the JDBC MessageStore in use with a larger number of sessions? Let's say 50+ sessions?
I currently struggle to use it because the connection pooling lib Proxool (which QFJ uses) is not maintained anymore and if I ran into issues then probably there would be no bug fixes.

Probably it would be not such a big deal to replace it with something better, e.g. HikariCP...?
Did someone already replace Proxool with something different?

Feedback is highly appreciated.

Thanks,
Chris.

**Comment by dparui** [comment, marked as answer]
Hi,

I know QFJ is free of Spring but here is what I have done using Spring for anyone interested.

I am using Spring JPA Repository instead of `Proxool`. I have my own implementation of `MessageStore` that calls the repository method for storing and retrieving message sequence numbers. This `MessageStore` implementation is returned by my implementation of `MessageStoreFactory`.  This spring bean is autowired in my implementation of `ThreadedSocketInitiator`.

Regards,
Diptanshu

**Reply by chrjohn** [reply]
Thanks for sharing. :)

**Reply by iavinash** [reply]
Hi @dparui 
I am also looking to get rid of `Proxool` as I am experiencing disconnections with `QuickfixJ `and I have raised a bug [here](https://github.com/quickfix-j/quickfixj/issues/795). Would you mind sharing your code snnipets?

**Comment by QwertGold** [comment]
We also use it with Spring, and create our JdbcStoreFactory this way.
```
JdbcStoreFactory storeFactory = new JdbcStoreFactory(settings);
storeFactory.setDataSource(dataSource);
```
When you set the Datasource explicitly, it is preferred over one created from settings. This allows us to set the Hikari pooled datasource from Spring.

**Reply by the-thing** [reply]
Hi @QwertGold,

I'm working on quickfixj HikariCP out of the box integration. Would you mind sharing which parameters do you explicitly set for com.zaxxer.hikari.HikariConfig?
